### PR TITLE
Abstraction of SQLite Handling

### PR DESCRIPTION
**Pull request by Pietfried**
- added init.sql script for initialization of database
- moved database handling from charge_point_configuration to charge_point
- implemented database_handler class for all operations on the database
- all database related operations now use the database_handler
- added new database functionalities to libocpp build examples
- added tests
- transactions will now be inserted and updated in the database
- incomplete transactions will now be stopped on startup by sending a StopTransaction.req for transactions without a meter_stop or time_end

Signed-off-by: pietfried <piet.goempel@pionix.de>

### CODE REVIEW

**Comment by Pietfried** [comment]
Should me merged with https://github.com/EVerest/everest-core/pull/120

File: lib/ocpp1_6/database_handler.cpp [hunk 1]
```diff
@@ -0,0 +1,601 @@
+
```
Comment (hikinggrass) [inline lib/ocpp1_6/database_handler.cpp:1]: Missing license header

File: lib/ocpp1_6/database_handler.cpp [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+        EVLOG_error << "Error closing database file: " << sqlite3_errmsg(this->db);
+    }
+}
+
+void DatabaseHandler::open_db_connection(int32_t number_of_connectors) {
+    if (sqlite3_open(this->db_path.c_str(), &this->db) != SQLITE_OK) {
+        EVLOG_error << "Error opening database: " << sqlite3_errmsg(db);
+        throw std::runtime_error("Could not open database at provided path.");
+    }
+    EVLOG_info << "Established connection to Database.";
```
Comment (hikinggrass) [inline lib/ocpp1_6/database_handler.cpp:31]: Does this (and similar log messages in this file) really have to be of loglevel INFO?

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+
```
Comment (hikinggrass) [inline tests/CMakeLists.txt:1]: empty line?

File: tests/database_tests.cpp [hunk 4]
```diff
@@ -0,0 +1,267 @@
+
```
Comment (hikinggrass) [inline tests/database_tests.cpp:1]: missing license header

**Review by hikinggrass — CHANGES_REQUESTED** [review]

**Pull request merged**